src/StepAdaptationController/src/StepAdaptationController.cpp

```diff
@@ -0,0 +1,541 @@
+
+// std
+#define NOMINMAX
+#include <algorithm>
+
+// yarp
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Value.h>
+
+// iDynTree
+#include <iDynTree/Core/EigenSparseHelpers.h>
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/SpatialAcc.h>
+#include <iDynTree/Core/Direction.h>
+
+#include <WalkingControllers/StepAdaptationController/StepAdaptationController.hpp>
+#include <WalkingControllers/YarpUtilities/Helper.h>
+#include <WalkingControllers/iDynTreeUtilities/Helper.h>
+
+typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> MatrixXd;
+using namespace WalkingControllers;
+
+StepAdaptationController::StepAdaptationController(const int &inputSize, const int &numberOfAllConstraints)
+    : m_inputSize(inputSize)
+    , m_numberOfConstraints(numberOfAllConstraints)
+    , m_xPositionsBuffer(2)
+    , m_yPositionsBuffer(2)
+    , m_zPositionsBuffer(3)
+    , m_zzPositionsBuffer(2)
+    , m_yawsBuffer(2)
+    , m_timesBuffer(2)
+    , m_zTimesBuffer(3)
+    , m_zzTimesBuffer(2)
+{
+    // instantiate the solver class
+    m_QPSolver = std::make_unique<OsqpEigen::Solver>();
+
+    //set the number of deceision variables of QP problem
+    m_QPSolver->data()->setNumberOfVariables(inputSize);
+
+    // set the number of all constraints includes inequality and equality constraints
+    m_QPSolver->data()->setNumberOfConstraints(numberOfAllConstraints);
+
+    m_QPSolver->settings()->setVerbosity(false);
+    m_QPSolver->settings()->setPolish(true);
+
+    m_constraintsMatrix.resize(numberOfAllConstraints, inputSize);
+    m_upperBound.resize(numberOfAllConstraints);
+    m_lowerBound.resize(numberOfAllConstraints);
+    m_gradient.resize(inputSize);
+
+    // set the constant elements of the constraint matrix
+    m_constraintsMatrix(0, 0) = 1;
+    m_constraintsMatrix(1, 1) = 1;
+    m_constraintsMatrix(0, 3) = 1;
+    m_constraintsMatrix(1, 4) = 1;
+    m_constraintsMatrix(6, 2) = 1;
```

Comment:
It is pretty weird that ``m_constraintsMatrix`` is resized according to ``numberOfAllCostraints`` and ``inputSize``, but then some values are hardcoded. What is ``numberOfAllConstraints < 7``?

If ``numberOfAllConstraints`` and ``InputSize`` are fixed, I would avoid passing them as parameters.

Reply:
At first I was thinking to have some more general formulation for step adaptation to have flexibility but at the end I couldn't to do that so I will fix it  as you suggested.

Reply:
Done!